### PR TITLE
ceph: update lib bucket provisioner

### LINKED ISSUE #6650: ObjectBucketClaim is not properly created using 1.5.0: resource name may not be empty
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
`ObjectBucketClaim` is not properly created.

After creating an `ObjectBucketClaim` using v1.5.0, Rook operator begins to loop, repeatedly creating multiple Ceph users and attempting to create the same bucket over and over. Only the first create bucket call succeeds, the others throw 409 Conflicts. During this time, the operator is showing the following error:

```
2020-11-17 03:37:16.783858 I | op-bucket-prov: getting storage class ""
2020-11-17 03:37:16.783923 E | op-bucket-prov: error getting class "", retrying. resource name may not be empty
```
Of course, as shown in the reproduction steps below, a StorageClass does in fact exist prior to creating an `ObjectBucketClaim`.

**Expected behavior:**
An ObjectBucketClaim is properly created

**How to reproduce it (minimal and precise):**

1. Spin up the Rook operator using Helm chart v1.5.0
2. Create a `CephCluster` using YAML manifest from examples directory
3. Create a `CephObjectStore` with 2/1 EC
4. Create a `rook-ceph.ceph.rook.io/bucket` `StorageClass`
5. Create a `ObjectBucketClaim` with the `spec.bucketName` specified and `spec.storageClassName` pointing to the previously created `StorageClass` (`rook-ceph-bucket

This error happens during a fresh install of v1.5.0. When downgraded to v1.4.7 (after wiping everything, including CRD's) things work fine.

**File(s) to submit**:

* Cluster CR (custom resource), typically called `cluster.yaml`, if necessary
* Operator's logs, if necessary
* Crashing pod(s) logs, if necessary

 To get logs, use `kubectl -n <namespace> logs <pod name>`
When pasting logs, always surround them with backticks or use the `insert code` button from the Github UI.
Read [Github documentation if you need help](https://help.github.com/en/articles/creating-and-highlighting-code-blocks).

**Environment**:
* OS (e.g. from /etc/os-release): Ubuntu 20.04.1 LTS (Focal Fossa)
* Kernel (e.g. `uname -a`): `Linux 5.4.0-52-generic #57-Ubuntu SMP Thu Oct 15 10:57:00 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux`
* Cloud provider or hardware configuration: Bare Metal Dedicated Server using private LAN
* Rook version (use `rook version` inside of a Rook Pod): v1.5.0
* Storage backend version (e.g. for ceph do `ceph -v`): ceph version 15.2.5 (2c93eff00150f0cc5f106a559557a58d3d7b6f1f) octopus (stable)
* Kubernetes version (use `kubectl version`): v1.19.4
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift): Bare Metal via Kubeadm
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/master/ceph-toolbox.html)): HEALTH_OK

### PR DESCRIPTION
**Pull request by BlaineEXE**
Update to the latest lib bucket provisioner code.
Fixes #6650

Modifies CRD for objectbucketclaims to fix an additional bug where an
ObjectBucket's 'ClaimRef' is lost due to the CRD validation being
specified incorrectly.

Does not reintroduce bug #6767 from previous fix for #6650

Signed-off-by: Blaine Gardner <blaine.gardner@redhat.com>

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #6650

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -79,3 +79,5 @@ replace (
 	k8s.io/sample-cli-plugin => k8s.io/sample-cli-plugin v0.19.3
 	k8s.io/sample-controller => k8s.io/sample-controller v0.19.3
 )
+
+replace github.com/kube-object-storage/lib-bucket-provisioner => github.com/BlaineEXE/lib-bucket-provisioner 7027296f348eaeb7a254ef7c5ef8d8f0fb1cda03
```
Comment (travisn) [inline go.mod:83]: This is just temporary until https://github.com/kube-object-storage/lib-bucket-provisioner/pull/200 merges, right?

Comment (BlaineEXE) [reply go.mod:83]: Yes. Mostly I want to see that CI will run against this. I think if you have a look at https://github.com/kube-object-storage/lib-bucket-provisioner/pull/200 that would be most helpful right now. I've tested the upgrade scenario now with success.

**Comment by BlaineEXE** [comment]
I want to add basic OBC validation to the upgrade test also to make sure this is definitely fixed and to catch any regressions in the future. I would expect this would add less than a minute to the test time.

**Review by travisn — APPROVED** [review]

**Comment by BlaineEXE** [comment]
The tests are failing yesterday and today with what appears to be the original bug #6650 when cleanup/deletion thinks the storage class is `""`. This issue was fixed a couple weeks ago in lib-bucket-provisioner and doesn't make a lot of sense.

This might suggest that k8s is not keeping the parameter as part of the OB in integration tests, but I don't see any issue with the CRDs for object bucket-related resources.

I cannot repro this condition locally on k8s v1.20, so I will see if v1.19.2 (the integration test version of k8s) lets me repro the error. Maybe there is a discrepancy between how v1.19 and v1.20 handle the API calls, but this is unlikely.

**Comment by BlaineEXE** [comment]
Finally tracked down the CI bug with a small fix in lib-bucket-provisioner to fix it. https://github.com/kube-object-storage/lib-bucket-provisioner/pull/201

**Pull request merged**